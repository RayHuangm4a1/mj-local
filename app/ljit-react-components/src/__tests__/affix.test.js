import React from 'react';
import { shallow, mount, } from 'enzyme';
import Affix from '../components/affix';

describe('affix', () => {
	beforeEach(function () {
		jest.mock('antd/lib/affix');
	});

	afterEach(function () {
		jest.resetModules();
	});

	it('should handle default props', () => {
		const {
			onChange,
		} = Affix.defaultProps;

		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const positionToTop = 10;
		const positionToBottom = 100;

		const wrapper = shallow(<Affix
			positionToTop={positionToTop}
			positionToBottom={positionToBottom}
		/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const positionToTop = 10;
		const positionToBottom = 100;
		const className = 'mock-class';
		const children = 'mock-children';
		const style = { position: 'absolute', width: '100px', };
		const onChange = () => {};

		const wrapper = mount(<Affix
			positionToTop={positionToTop}
			positionToBottom={positionToBottom}
			style={style}
			className={className}
			onChange={onChange}
		>
			{children}
		</Affix>);

		expect(wrapper.props().positionToTop).toBe(positionToTop);
		expect(wrapper.props().positionToBottom).toBe(positionToBottom);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().style).toMatchObject(style);
		expect(wrapper.props().children).toBe(children);
		expect(wrapper.props().onChange).toEqual(onChange);
	});

	it('should handle change event', () => {
		const value = 'mock-value';
		const onChange = jest.fn();
		const event = {
			preventDefault: () => {},
			target: { value, },
		};
		const wrapper = shallow(<Affix onChange={onChange} />);

		wrapper.simulate('change', event);
		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toBeCalledWith(event);
	});
});
