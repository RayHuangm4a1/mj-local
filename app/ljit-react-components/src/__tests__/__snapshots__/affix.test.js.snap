// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`affix should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Affix
    onChange={[Function]}
    positionToBottom={100}
    positionToTop={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "className": "ljit-affix",
      "offsetBottom": 100,
      "offsetTop": 10,
      "onChange": [Function],
      "style": undefined,
      "target": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "className": "ljit-affix",
        "offsetBottom": 100,
        "offsetTop": 10,
        "onChange": [Function],
        "style": undefined,
        "target": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
